src/main/java/org/gridsuite/sensitivityanalysis/server/repositories/SensitivityAnalysisResultRepository.java

```diff
@@ -136,7 +137,9 @@ private int apply(AnalysisResultEntity sas, Collection<String> funcIds, Collecti
 
             int ci = sar.getContingencyIndex();
             ContingencyEmbeddable c = ci < 0 ? null : cs.get(ci);
-            if ((ci < 0) == beforeOverAfter) {
+            if ((ci < 0) != (allContingencyIds == null)) {
```

Comment:
checkstyle UnnnecessaryParentheses is raised here (with powsybl checkstyle configuration file)

Reply:
Uh, wouldn't have guessed, as it is not my habit of using surnumeral parentheses.
But here I wanted to emphasize that the topmost condition is an equality between booleans,
so I addes parentheses around the inequality test.

Reply:
Done